### PR TITLE
Fix for postgres deployment CR for openshift

### PR DESCRIPTION
**Pull request by Rajagopalan-Ranganathan**
## JIRA ID
 https://issues.redhat.com/browse/KEYCLOAK-16864

## Additional Information
Init containers change that was added would work only
with plain Kubernetes. Code changes made to detect if
the underlying cluster is openshift or Kubernetes and then to add the init containers
part based on that.

   Follow steps here: https://github.com/keycloak/keycloak-operator#local-development for running the operator locally
>  verify that pods are created and running (keycloak and postgresql)
>  Verify that deployment config of postgres doesnt contain initcontainers sections for openshift, but has the same for plain kuberntes
> verify key cloak works fine - sanity check

## Checklist:
- [X] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [ ] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

**Comment by Rajagopalan-Ranganathan** [comment]
@slaskawi @akoserwal

File: go.mod [hunk 1]
```diff
@@ -15,6 +15,7 @@ require (
 	k8s.io/apimachinery v0.18.3
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6
+	k8s.io/utils v0.0.0-20200414100711-2df71ebbae66
```
Comment (slaskawi) [inline go.mod:18]: This change is not related to this Pull Request. Please remove it.

File: pkg/controller/keycloak/keycloak_reconciler_test.go [hunk 2]
```diff
@@ -581,6 +583,7 @@ func TestKeycloakReconciler_Test_Setting_Resources(t *testing.T) {
 	stateManager.SetState(common.GetStateFieldName(ControllerName, monitoringv1.ServiceMonitorsKind), true)
 	stateManager.SetState(common.GetStateFieldName(ControllerName, grafanav1alpha1.GrafanaDashboardKind), true)
 	stateManager.SetState(common.RouteKind, true)
+	stateManager.SetState(common.OpenShiftAPIServerKind, false)
```
Comment (slaskawi) [inline pkg/controller/keycloak/keycloak_reconciler_test.go:586]: This one is probably not necessary. The code should assume it's `false` if it's not set.

File: pkg/model/postgresql_deployment.go [hunk 3]
```diff
@@ -169,6 +149,35 @@ func PostgresqlDeployment(cr *v1alpha1.Keycloak) *v13.Deployment {
 			},
 		},
 	}
+
+	//If not openshift append the initcontainer part
```
Comment (slaskawi) [inline pkg/model/postgresql_deployment.go:153]: This comment should probably go away. It's quite obvious what this code does.

**Review by slaskawi — CHANGES_REQUESTED** [review]
Great work @Rajagopalan-Ranganathan !

I found only a few, really minor comments. Please fix them and I'll merge this one.

File: go.mod (hunk 1 above)
Comment (Rajagopalan-Ranganathan) [reply go.mod:18]: Agreed. Removing the same.

File: pkg/controller/keycloak/keycloak_reconciler_test.go (hunk 2 above)
Comment (Rajagopalan-Ranganathan) [reply pkg/controller/keycloak/keycloak_reconciler_test.go:586]: Agreed. Removed the same.

File: pkg/model/postgresql_deployment.go (hunk 3 above)
Comment (Rajagopalan-Ranganathan) [reply pkg/model/postgresql_deployment.go:153]: Agreed, removed the comment.

**Review by slaskawi — APPROVED** [review]

**Comment by slaskawi** [comment]
Integrated, thanks @Rajagopalan-Ranganathan
